src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,428 @@
+        alignItems: 'baseline',
+        paddingBottom: 5,
+        marginTop: 10,
+        marginBottom: 5,
+        borderBottom: `1px solid ${colors.dark(0.35)}`,
+      },
+    },
+    [
+      div({ style: { margin: '5px', display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            'aria-label': 'delete criteria',
+            onClick: () => {
+              deleteCriteria(criteria);
+            },
+          },
+          [icon('minus-circle', { size: 24, style: { color: colors.danger() } })]
+        ),
+        div({ style: { marginLeft: 5 } }, [
+          Utils.cond(
+            [
+              'domainType' in criteria,
+              () => {
+                const domainCriteria = criteria as DomainCriteria;
+                return h(Fragment, [strong([`${domainCriteria.domainType.category}:`]), ` ${domainCriteria.name}`]);
+              },
+            ],
+            [
+              'listType' in criteria,
+              () => {
+                const listCriteria = criteria as ProgramDataListCriteria;
+                return h(Fragment, [strong([`${criteria.name}:`]), ` ${listCriteria.value.name}`]);
+              },
+            ],
+            [
+              'rangeType' in criteria,
+              () => {
+                const rangeCriteria = criteria as ProgramDataRangeCriteria;
+                return h(Fragment, [strong([`${criteria.name}:`]), ` ${rangeCriteria.low} - ${rangeCriteria.high}`]);
+              },
+            ],
+            [Utils.DEFAULT, () => div(['Unknown criteria type'])]
+          ),
+        ]),
+      ]),
+      `Count: ${criteria.count}`,
+    ]
+  );
+};
+
+export const renderCriteriaView = (deleteCriteria: (criteria: AnyCriteria) => void) => (criteria: AnyCriteria) =>
+  h(CriteriaView, { deleteCriteria, criteria, key: criteria.id });
+
+let criteriaCount = 1;
+const selectDomainCriteria = (domainType: DomainType): DomainCriteria => {
+  // This needs to be replaced with a UI that lets users select the criteria they want from
+  // the list of concepts for this domain.
+  return {
+    domainType,
+    name: domainType.values[0],
+    id: criteriaCount++,
+    // Need to call the API service to get the count for this criteria.
+    count: 100,
+  };
+};
+
+const createDefaultListCriteria = (listType: ProgramDataListType): ProgramDataListCriteria => {
+  return {
+    listType,
+    name: listType.name,
+    id: criteriaCount++,
+    count: 100,
+    value: listType.values[0],
+  };
+};
+
+const createDefaultRangeCriteria = (rangeType: ProgramDataRangeType): ProgramDataRangeCriteria => {
+  return {
+    rangeType,
+    name: rangeType.name,
+    id: criteriaCount++,
+    count: 100,
+    low: rangeType.min,
+    high: rangeType.max,
+  };
+};
+
+export function createCriteriaFromType(type: CriteriaType): AnyCriteria {
+  return (
+    Utils.condTyped<AnyCriteria>(
+      ['category' in type, () => selectDomainCriteria(type as DomainType)],
```

Comment:
For naming consistency, did you think about calling this function `createDefaultDomainCriteria`?

Reply:
I named them differently because the inline editors will need to have some initial state to show, whereas a domain critera has no initial state but instead shows a selector UI. It occurs to e now that in our current UI spec `selectDomainCriteria` can return more than one criteria (or none) but change that can wait until we add the domain criteria selector.